pkg/nto/performanceprofile_test.go

```diff
@@ -0,0 +1,662 @@
+		assert.Equal(t, testCase.expectedStatus, exist)
+	}
+}
+
+func TestPerformanceProfileGet(t *testing.T) {
+	testCases := []struct {
+		testPerformanceProfile *Builder
+		expectedError          error
+	}{
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          nil,
+		},
+		{
+			testPerformanceProfile: buildInValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          fmt.Errorf("PerformanceProfile's name is empty"),
+		},
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:          fmt.Errorf("performanceprofiles.performance.openshift.io \"default\" not found"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		performanceProfileObj, err := testCase.testPerformanceProfile.Get()
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, performanceProfileObj.Name, testCase.testPerformanceProfile.Definition.Name)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestPerformanceProfileCreate(t *testing.T) {
+	testCases := []struct {
+		testPerformanceProfile *Builder
+		expectedError          string
+	}{
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          "",
+		},
+		{
+			testPerformanceProfile: buildInValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          "PerformanceProfile's name is empty",
+		},
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:          "resourceVersion can not be set for Create requests",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testPerformanceProfileBuilder, err := testCase.testPerformanceProfile.Create()
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testPerformanceProfileBuilder.Definition.Name, testPerformanceProfileBuilder.Object.Name)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		}
+	}
+}
+
+func TestPerformanceProfileDelete(t *testing.T) {
+	testCases := []struct {
+		testPerformanceProfile *Builder
+		expectedError          error
+	}{
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          nil,
+		},
+		{
+			testPerformanceProfile: buildInValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          fmt.Errorf("PerformanceProfile's name is empty"),
+		},
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:          fmt.Errorf("PerformanceProfile cannot be deleted because it does not exist"),
```

Comment:
Please change Delete method. 
We have to return nil and set Object to nil if actual object doesn't exist on cluster.